Convert all uses of RefPtr<WebCore::StaticRange> to Ref<WebCore::StaticRange>.

MoeBazziGIT opened the pull request:
#### dd6a9e3510cec677419e5b242161d88fdeb0e077
<pre>
Convert all uses of RefPtr&lt;WebCore::StaticRange&gt; to Ref&lt;WebCore::StaticRange&gt;.
</pre>
----------------------------------------------------------------------
#### 30ec2970898cea0ee75c137ec6bf1ad3475d1f6b
<pre>
Default initialize InputEventInit IDL members
</pre>
----------------------------------------------------------------------
#### 42b41cb804fdb56afc99d70fd0bf19c1577b03f4
<pre>
Support InputEventInit.{inputType, dataTransfer, isComposing, targetRanges}
<a href="https://bugs.webkit.org/show_bug.cgi?id=170416">https://bugs.webkit.org/show_bug.cgi?id=170416</a>

Reviewed by NOBODY (OOPS!).

This adds support for more initialization parameters for InputEvent, such as dataTransfer and targetRanges.

* LayoutTests/imported/w3c/web-platform-tests/uievents/constructors/inputevent-constructor-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/uievents/constructors/inputevent-constructor.html:
* Source/WebCore/dom/InputEvent.cpp:
(WebCore::InputEvent::InputEvent):
* Source/WebCore/dom/InputEvent.h:
* Source/WebCore/dom/InputEvent.idl:
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/dd6a9e3510cec677419e5b242161d88fdeb0e077

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/74492 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/53921 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/27303 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/78890 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/25730 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/76609 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/63054 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/1706 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/58551 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/16850 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/77559 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/48699 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/64055 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/38953 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/45754 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/21552 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/24063 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/67104 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/21897 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/80400 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/1809 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/1064 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/66822 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/1957 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/64073 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/66107 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/10043 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/8199 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/1773 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/4561 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/1802 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/1790 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/1809 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

whsieh commented on Source/WebCore/dom/InputEvent.h:
```diff
@@ -42,6 +42,8 @@ class InputEvent final : public UIEvent {
         bool isComposing { false }; // input method
         String inputType;
+        Vector<RefPtr<StaticRange>> targetRanges;
+        RefPtr<DataTransfer> dataTransfer;
```
Nit - can we keep these in an order that's consistent with the IDL? (i.e. `dataTransfer` and then `targetRanges`)

whsieh reviewed (commented):
Thanks for the patch! A couple of comments:

1. Move the summary describing the change in the commit message, below the bug number and change the title to be consistent with the bugzilla bug. i.e.,:
```
Support InputEventInit.{inputType, dataTransfer, isComposing, targetRanges}
https://bugs.webkit.org/show_bug.cgi?id=170416

Reviewed by NOBODY (OOPS!).

This adds support for more initialization parameters for InputEvent, such as dataTransfer and targetRanges.

* Source/WebCore/dom/InputEvent.cpp:
(WebCore::InputEvent::InputEvent):
* Source/WebCore/dom/InputEvent.h:
* Source/WebCore/dom/InputEvent.idl:
```
2. We should add a new test (or modify existing tests) to reflect this change. The existing WPT `imported/w3c/web-platform-tests/uievents/constructors/inputevent-constructor.html` seems pertinent.

It seems like [the spec](https://www.w3.org/TR/uievents/#idl-inputeventinit) doesn't include these two arguments, but testing in Chrome and Firefox, I can see that they both allow you to specify `targetRanges` and `dataTransfer` in `InputEventInit`, so this change would align with Firefox/Chrome. It seems that the spec just needs to be updated.

MoeBazziGIT committed:
Default initialize InputEventInit IDL members

MoeBazziGIT committed:
Support InputEventInit.{inputType, dataTransfer, isComposing, targetRanges}
https://bugs.webkit.org/show_bug.cgi?id=170416

Reviewed by NOBODY (OOPS!).

This adds support for more initialization parameters for InputEvent, such as dataTransfer and targetRanges.

* LayoutTests/imported/w3c/web-platform-tests/uievents/constructors/inputevent-constructor-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/uievents/constructors/inputevent-constructor.html:
* Source/WebCore/dom/InputEvent.cpp:
(WebCore::InputEvent::InputEvent):
* Source/WebCore/dom/InputEvent.h:
* Source/WebCore/dom/InputEvent.idl:

MoeBazziGIT committed:
Convert all uses of RefPtr<WebCore::StaticRange> to Ref<WebCore::StaticRange>.
